h2o-extensions/xgboost/src/main/java/hex/tree/xgboost/rabit/RabitTrackerH2O.java

```diff
@@ -20,7 +20,7 @@
 
     private Map<String, String> envs = new HashMap<>();
 
-    private RabitTrackerH2OThread trackerThread;
+    private volatile RabitTrackerH2OThread trackerThread;
```

Comment:
Red from both the thread and this class. Better guarantee single place of existence.